File: 1886-minimum-limit-of-balls-in-a-bag/1886-minimum-limit-of-balls-in-a-bag.py
class Solution:
    def minimumSize(self, nums: List[int], maxOperations: int) -> int:
        def oprCount(val):
            temp = 0
            for i in range(len(nums)):
                temp += ceil(nums[i] / val ) -1
            return temp
        minBall = 1
        maxBall = max(nums)
        while(minBall < maxBall):
            mid = (minBall + maxBall)//2
            if ( oprCount(mid) <= maxOperations ) :
                maxBall = mid 
            else : 
                minBall = mid + 1
        return maxBall        

File: 1886-minimum-limit-of-balls-in-a-bag/1886-minimum-limit-of-balls-in-a-bag.java
class Solution {
    public int bagCounter( int val ,int [] nums) 
    {
        int temp = 0;
        for( int i = 0 ; i < nums.length ; i++)
        {
            temp += Math.ceil((double)nums[i] / val) - 1;
        }
        return temp ;
    }
    public int minimumSize(int[] nums, int maxOperations) {
        int minBalls = 1;
        Arrays.sort(nums);
        int maxBalls = nums[nums.length - 1];
        while ( minBalls < maxBalls )
        {
            int mid = (minBalls + maxBalls) / 2;
            if( bagCounter(mid,nums) <= maxOperations)
            {
                maxBalls = mid;
            }
            else
            {
                minBalls = mid + 1;
            }
        }
        return maxBalls;
    }
}